tests/scripts/all.sh

```diff
@@ -3503,6 +3503,46 @@ component_test_malloc_0_null () {
     tests/ssl-opt.sh -e 'proxy'
 }
 
+support_test_aesni () {
+    # require an x86_64 target
+    gcc -v 2>&1 | grep Target | grep -q x86_64
+}
+
+component_test_aesni () { # ~ 60s
+    # This tests the two AESNI implementations (intrinsics and assembly), and also the plain C
+    # fallback. It also tests the logic that is used to select which implementation(s) to build.
+    #
+    # This test does not require the host to have support for AESNI (if it doesn't, the run-time
+    # AESNI detection will fallback to the plain C implementation, so the tests will instead
+    # exercise the plain C impl).
+
+    msg "build: default config with different AES implementations"
+    scripts/config.py set MBEDTLS_AESNI_C
+    scripts/config.py set MBEDTLS_HAVE_ASM
+
+    # test the intrinsics implementation
+    msg "AES tests, test intrinsics"
+    make clean
+    make test programs/test/selftest CC=gcc CFLAGS='-Werror -Wall -Wextra -mpclmul -msse2 -maes'
+    # check that we built intrinsics - this should be used by default when supported by the compiler
+    ./programs/test/selftest | grep "AESNI code" | grep -q "intrinsics" || false "intrinsics not built when supported"
```

Comment:
Using a conditional construct like `foo || bar` makes bash not react to the failure of `foo` **or `bar`**. It's a quirk of sh/bash programming. (Yes, we should rewrite this script in Python. Yes, it's on my tech debt list, been there since about 2018.) Thus, even if the `bar` branch is invoked, the test component will pass.

The simple fix is to just not get a custom message. The report will still point to the failing command (`grep -q "intrinsics"`) so it's enough to figure out what failed without undue effort.
```suggestion
    ./programs/test/selftest | grep "AESNI code" | grep -q "intrinsics"
```

Same with the other `|| false …` below.